pmlc/rt/runtime.cc

```diff
@@ -19,4 +19,14 @@ ScopedCurrentDevice::~ScopedCurrentDevice() { currentDevice = nullptr; }
 
 std::shared_ptr<Device> Device::currentUntyped() { return currentDevice; }
 
+void Runtime::initRegisteredRuntimes() {
+  static std::once_flag inited;
+  std::call_once(inited, []() {
+    for (auto &[id, runtime] : getRuntimeMap()) {
+      std::call_once(runtime->initialized,
+                     [runtime = runtime.get()]() { runtime->init(); });
+    }
```

Comment:
Seems like we need to skip runtimes that fail to initialize. Or we could remove them from the map if they fail. Thoughts?